### PR TITLE
Fixes #1775 Soft delete bug

### LINKED ISSUE #1775: Soft delete bug
I disabled "Soft delete feature" in ACP -&gt; General configuration

But I can still soft delete threads and posts... I can see "Soft dete thread" in forum and when I use this, the thread is soft deleted

Is it a bug?

Original thread: [Soft delete bug](http://community.mybb.com/thread-165872.html)

### PR DESCRIPTION
**Pull request by PaulBender**
Fixes #1775 

Also cleared up some permission errors and moved hardcoded HTML soft delete/restore moderation options to templates.

### CODE REVIEW

File: install/resources/upgrade32.php [hunk 1]
```diff
@@ -50,6 +50,7 @@ function upgrade32_dbchanges()
 	$db->delete_query("forumpermissions", "fid NOT IN(SELECT fid FROM {$db->table_prefix}forums)");
 
 	$db->update_query("settings", array('optionscode' => 'select\r\n0=No CAPTCHA\r\n1=MyBB Default CAPTCHA\r\n2=reCAPTCHA\r\n3=Are You a Human\r\n4=NoCAPTCHA reCAPTCHA'), "name='captchaimage'");
+	$db->update_query("settings", array('title' => 'Enable Soft Delete for Users', 'description' => 'If enabled, posts and threads deleted by users will be hidden and can be restored by moderators. Otherwise, these posts and threads will be deleted permanently.'), "name='soft_delete'");
```
Comment (Sama34) [inline install/resources/upgrade32.php]: Doesn't the upgrade script updates this automatically?

Comment (JN-Jones) [inline install/resources/upgrade32.php]: From what I've seen it should even update the optionscode in the line above. Though the code in the `sync_settings` function is a bit weird it should do it.

Comment (PaulBender) [inline install/resources/upgrade32.php]: I've removed it.

**Comment by Cu8eR** [comment]
@PaulBender - there is a conflict

**Comment by PaulBender** [comment]
Fixed

**Comment by Cu8eR** [comment]
Looks good

**Comment by JN-Jones** [comment]
When a moderator has the permission to soft delete threads but not posts (may also happen with hard delete from a look at the code) the Quick Delete button shows but will throw an error as the codeblock from line 115 on checks only post permissions. However it's possible to (soft) delete the thread via the mod dropdown in the thread overview.

**Comment by JN-Jones** [comment]
Rest seems to work, at least I haven't found any other things.

**Comment by PaulBender** [comment]
Fixed

**Comment by Destroy666x** [comment]
What about these?
https://github.com/PaulBender/mybb/blob/fix-1775/editpost.php#L292
https://github.com/PaulBender/mybb/blob/fix-1775/editpost.php#L347
Shouldn't they be changed too?

**Comment by PaulBender** [comment]
They've been changed

**Comment by Destroy666x** [comment]
I still think there are some inconsistencies for mod permissions like `canviewdeleted` (in search.php for example), but this PR looks good and can be merged, the rest can be fixed in 1.8.5.

**Comment by PaulBender** [comment]
Fixed merge conflict.

**Pull request merged**